packages/mui-joy/src/styles/defaultTheme.ts

```diff
@@ -81,45 +134,209 @@ type BasePaletteRange = 50 | 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900
+      color: 'var(--joy-palette-text-heading)',
+    },
+    h4: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-medium)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-xl)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-normal)',
+      textTransform: 'capitalize',
+      color: 'var(--joy-palette-text-heading)',
+    },
+    h5: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-regular)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-lg)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-normal)',
+      textTransform: 'capitalize',
+      color: 'var(--joy-palette-text-heading)',
+    },
+    headingSubtitle: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-regular)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-lg)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-normal)',
+      textTransform: 'capitalize',
+      color: 'var(--joy-palette-text-content)',
+    },
+    body: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-regular)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-md)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-normal)',
+      textTransform: 'capitalize',
+      color: 'var(--joy-palette-text-content)',
+    },
+    caption: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-regular)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-sm)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-normal)',
+      textTransform: 'capitalize',
+      color: 'var(--joy-palette-text-content)',
+    },
+    detail: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-regular)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-xs)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-normal)',
+      textTransform: 'capitalize',
+      color: 'var(--joy-palette-text-detail)',
+    },
+    headingIntro: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-extraBold)' as React.CSSProperties['fontWeight'],
+      fontSize: 'var(--joy-fontSize-md)',
+      lineHeight: 'var(--joy-lineHeight-normal)',
+      letterSpacing: 'var(--joy-letterSpacing-lg)',
+      textTransform: 'uppercase',
+      color: 'var(--joy-palette-text-headingIntro)',
+    },
+    overline: {
+      fontFamily: 'var(--joy-fontFamily-sans)',
+      fontWeight: 'var(--joy-fontWeight-extraBold)' as React.CSSProperties['fontWeight'],
```

Comment:
Should we accept here string too rather than cast everywhere?

Reply:
Yep, that is much clearer.

Reply:
I tried but it affects another part. I think I will move forward with typecasting for now.

![image](https://user-images.githubusercontent.com/18292247/140613379-eb9b62c4-9f22-4146-9f68-41d0f8acc5ac.png)